17728: test-infra: Add device_registration

CyrusJavan opened the pull request:
Add device_registration resource to the relevant test-infra scripts.

Also add module _cisco-csr_ to bring up the infra for the cloudWAN
tests.

Also added SKIP env vars for the reosurces that are in the provider
but have not been added to the test-infra yet.

arvindsree commented on aviatrix/resource_aviatrix_spoke_gateway.go:
```diff
@@ -593,20 +600,22 @@ func resourceAviatrixSpokeGatewayUpdate(d *schema.ResourceData, meta interface{}
-		o, n := d.GetChange("enable_nat")
-		if o == "yes" && n == "no" {
+
+		enableNat := d.Get("enable_nat").(bool)
```
why change this logic? If nat is already enabled, you dont need to enable it again

arvindsree replied:
nevermind..it is inside d.haschange

arvindsree commented on test-infra/outputs.tf:
```diff
@@ -92,5 +96,14 @@ output "AWS_VPC_ID2" {
+}
+
+output "AWS_REGION" {
+  value = var.aws_region1
```
looks like you are not using these in the vpc. You could use a data source to get current region

CyrusJavan committed:
17728: test-infra: Add device_registration

Add device_registration resource to the relevant test-infra scripts.

Also add module _cisco-csr_ to bring up the infra for the cloudWAN
tests.

Also added SKIP env vars for the reosurces that are in the provider
but have not been added to the test-infra yet.
